src/webots/sound/WbPicoTextToSpeech.cpp

```diff
@@ -351,6 +351,10 @@ qint16 *WbPicoTextToSpeech::generateBufferFromText(const QString &text, int *siz
         buffer = (qint16 *)realloc(buffer, bufferSize);
         if (!buffer)  // re-allocation failed, this is required otherwise CppCheck raises an error
           free(previousBuffer);
+        else {
+          gError = QString("Cannot re-allocate buffer.");
+          return NULL;
+        }
```

Comment:
This code seems wrong (and it is still failing at cppcheck on macOS https://github.com/cyberbotics/webots/pull/2327/checks?check_run_id=1259146248 ).

If `buffer` is not NULL, then the allocation didn't fail so we should neither return nor print an error.

Reply:
I fixed this in  #2347

Reply:
You're completely right, fixing this right now.
Strangely, the cppcheck did't fails my previous PRs today: https://github.com/cyberbotics/webots/runs/1257659737#step:4:45